utils/utils.go

```diff
@@ -48,8 +48,12 @@ func CustomizeDeployment(deploy *appsv1.Deployment, ba common.BaseComponent) {
 	}
 
 	dp := ba.GetDeployment()
-	if dp != nil && dp.GetDeploymentUpdateStrategy() != nil {
-		deploy.Spec.Strategy = *dp.GetDeploymentUpdateStrategy()
+	if dp != nil {
+		if dp.GetDeploymentUpdateStrategy() != nil {
+			deploy.Spec.Strategy = *dp.GetDeploymentUpdateStrategy()
+		} else {
+			deploy.Spec.Strategy = appsv1.DeploymentStrategy{Type: appsv1.RollingUpdateDeploymentStrategyType}
```

Comment:
The user could remove `spec.deployment` altogether. We would still have a Deployment, but with old/removed strategy. We should set the default strategy above the `dp != nil` check. It might be a good idea to add a test for this scenario as well. Make sure to specify a non-default update strategy beforehand, so we can confirm that it in fact went back to default. Sorry to pile on more requirements :) 

This is not an issue for StatefulSet, since user has to explicitly specify `spec.statefulSet` in order to create StatefulSet.